Re-diffed for 2.6.14.7 by Kerin Millar <kerframil@gmail.com>:

From: Simon Vogl <simon.vogl@researchstudios.at>
Date: Fri, 24 Feb 2006 21:04:09 +0000 (-0800)
Subject: [PATCH] cfi: init wait queue in chip struct
X-Git-Url: http://www.kernel.org/git/?p=linux/kernel/git/chrisw/linux-2.6.15.y.git;a=commitdiff;h=3bf8ce11c70ef5dc2298c7fff4a744c48e87b9f0

[PATCH] cfi: init wait queue in chip struct

Fix a kernel oops for Intel P30 flashes, where the wait queue head was not
initialized for the flchip struct, which in turn caused a crash at the
first read operation.

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Cc: <stable@kernel.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---

--- linux-2.6.14.7.orig/drivers/mtd/chips/cfi_cmdset_0001.c	2005-10-27 19:02:08.000000000 -0500
+++ linux-2.6.14.7/drivers/mtd/chips/cfi_cmdset_0001.c	2006-03-07 05:23:23.000000000 -0500
@@ -379,6 +379,7 @@
 		cfi->chips[i].buffer_write_time = 1<<cfi->cfiq->BufWriteTimeoutTyp;
 		cfi->chips[i].erase_time = 1<<cfi->cfiq->BlockEraseTimeoutTyp;
 		cfi->chips[i].ref_point_counter = 0;
+		init_waitqueue_head(&(cfi->chips[i].wq));
 	}		
 
 	map->fldrv = &cfi_intelext_chipdrv;
